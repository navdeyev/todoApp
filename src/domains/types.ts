import {RouterState} from 'react-router-redux';

import {ITheme} from 'components/Styled/themes';
import {MediaType} from 'components/Styled/utils';
import {LoadingStates} from 'domains/loadingStates';
import {ITodo} from 'domains/todos/todosTypes';
import {IWindowDimensions} from 'domains/window/windowTypes';

export interface ITodosState {
    loadingState: LoadingStates,
    todos: ITodo[],
    selectedTodoId: string
}

export interface IWindowState {
    windowDimensions: IWindowDimensions,
    media: MediaType
}

export interface IThemeState {
    theme: ITheme
}

export interface IAppState {
    routing: RouterState,
    todosState: ITodosState,
    windowState: IWindowState,
    themeState: IThemeState
}
