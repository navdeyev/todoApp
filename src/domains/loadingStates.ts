export enum LoadingStates {
    NOT_STARTED = 'NOT_STARTED',
    LOADING = 'LOADING',
    COMPLETE = 'COMPLETE',
}