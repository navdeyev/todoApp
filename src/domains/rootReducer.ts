import {routerReducer} from 'react-router-redux';
import {combineReducers} from 'redux';

import {IAppState} from './types';

import todosReducer from 'domains/todos/todosReducer';
import windowReducer from 'domains/window/windowReducer';

const rootReducer = combineReducers<IAppState>({
    routing: routerReducer,
    todosState: todosReducer,
    windowState: windowReducer
});

export default rootReducer;