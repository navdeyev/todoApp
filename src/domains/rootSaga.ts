import {all, fork} from 'redux-saga/effects';

import todosSaga from './todos/todosSaga';

import {IServiceMap} from 'service/services';

function* rootSaga(serviceMap: IServiceMap) {
    yield all([
        fork(todosSaga, serviceMap)
    ]);
}

export default rootSaga;