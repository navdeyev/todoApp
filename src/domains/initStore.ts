import {applyMiddleware, createStore, Store} from 'redux';
import {composeWithDevTools} from 'redux-devtools-extension'
import thunk from 'redux-thunk';

import rootReducer from './rootReducer';
import {IAppState} from './types';

const initStore = (): Store<IAppState> => {
    const middlewares = [
        thunk
    ];

    const enhancer = process.env.NODE_ENV === 'development'
        ? composeWithDevTools(applyMiddleware(...middlewares))
        : applyMiddleware(...middlewares);

    return createStore<IAppState>(rootReducer, enhancer);
};

export default initStore;