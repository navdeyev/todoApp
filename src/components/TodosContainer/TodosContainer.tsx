import * as React from 'react';
import {connect} from 'react-redux';
import {ActionCreator, AnyAction} from 'redux';

import {LoadingStates} from 'domains/loadingStates';
import * as todosActions from 'domains/todos/todosActions';
import {IAppState, ITodo} from 'domains/types';

import LoadingIndicator from 'components/LoadingIndicator/LoadingIndicator';

interface IProps {
    loadingState: LoadingStates,
    todos: ITodo[],
    loadTodos: ActionCreator<AnyAction>
}

export class TodosContainer extends React.Component<IProps> {

    public componentDidMount() {
        this.props.loadTodos();
    }

    public render() {
        const {loadingState, todos} = this.props;
        return (
            <LoadingIndicator loadingState={loadingState}>
                <div>{JSON.stringify(todos)}</div>
            </LoadingIndicator>
        );
    }

}

const mapStateToProps = (state: IAppState) => {
    return {
        loadingState: state.todosState.loadingState,
        todos: state.todosState.todos
    }
};

const mapDispatchToProps = {
    loadTodos: todosActions.loadTodos
};

export default connect(mapStateToProps, mapDispatchToProps)(TodosContainer);
