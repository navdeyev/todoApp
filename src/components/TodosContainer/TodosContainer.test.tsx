import {shallow} from 'enzyme';
import * as React from 'react';

import {ITodo, TodoStatus} from 'domains/types';
import {TodosContainer} from './TodosContainer';

describe('TodosContainer', () => {

    it('renders a component', () => {
        const todosArray: ITodo[] = [{
            goal: 'Goal',
            id: 'some-id',
            status: TodoStatus.NOT_STARTED,
            steps: [],
            title: 'Title'
        }];

        const props = {
            loadTodos: jest.fn(),
            todos: todosArray
        };
        const component = shallow(<TodosContainer {...props} />);
        expect(component.getElement()).toMatchSnapshot();
    });

});