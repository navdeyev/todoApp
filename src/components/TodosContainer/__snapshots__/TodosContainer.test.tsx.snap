// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodosContainer renders a component 1`] = `
<LoadingIndicator
  loadingState="COMPLETE"
>
  <TodosList
    todos={
      Array [
        Object {
          "goal": "Goal",
          "id": "some-id",
          "status": "PENDING",
          "steps": Array [],
          "title": "Title",
        },
      ]
    }
  />
</LoadingIndicator>
`;
